Tests/UpstreamIntegrationTests/util/XCTestCase+AnyCodableAsserts.swift

```diff
@@ -602,22 +602,68 @@ extension XCTestCase {
 
     // MARK: - Test setup and output helpers
     /// Applies the provided regex pattern to the text and returns all the capture groups from the regex pattern
-    private func getCapturedRegexGroups(text: String, regexPattern: String, file: StaticString = #file, line: UInt = #line) -> [String?] {
+    private func getCapturedRegexGroups(text: String, regexPattern: String, file: StaticString = #file, line: UInt = #line) -> [String] {
         do {
             let regex = try NSRegularExpression(pattern: regexPattern)
             let matches = regex.matches(in: text,
                                         range: NSRange(text.startIndex..., in: text))
-            return matches.map { match in
-                return (0..<match.numberOfRanges).map {
-                    let rangeBounds = match.range(at: $0)
+            var captureResult: [String] = []
+            for match in matches {
+                var capturedStrings: [String?] = []
+                // [(matched string), (capture group 0), (capture group 1), etc.]
+                for rangeIndex in 0..<match.numberOfRanges {
+                    let rangeBounds = match.range(at: rangeIndex)
                     guard let range = Range(rangeBounds, in: text) else {
-                        return ""
+                        capturedStrings.append(nil)
+                        continue
                     }
-                    return String(text[range])
+                    capturedStrings.append(String(text[range]))
                 }
-            }.map {
-                $0.last
+                // If there is a match it should be impossible for there to be no elements
```

Comment:
In your case it seems like the first element might be nil, any reason for adding that element instead of keeping the array empty?

Either way, it is safer to just check for ! campturedString.isEmpty right before these operations as per docs (https://developer.apple.com/documentation/swift/array/removefirst())